File: leetcode/771.py
"""
Runtime: 32 ms, faster than 69.54% of Python3 online submissions for Jewels and Stones.
Memory Usage: 14.2 MB, less than 44.84% of Python3 online submissions for Jewels and Stones.
"""
from collections import Counter


class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        s, j = Counter(stones), Counter(jewels)
        return sum([s[jewel] for jewel in j])

File: leetcode/771.cpp
// Runtime: 0 ms, faster than 100.00% of C++ online submissions for Jewels and Stones.
// Memory Usage: 6.1 MB, less than 96.01% of C++ online submissions for Jewels and Stones.
#include <string>

class Solution {
 public:
  int numJewelsInStones(string jewels, string stones) {
    int numJewels = 0;
    for (string::const_iterator it = stones.begin(); it != stones.end(); ++it) {
      if (jewels.find(*it) != std::string::npos) {
        numJewels += 1;
      }
    }

    return numJewels;
  }
};
